File: kotlin-algo/src/selection_sort.kt
import kotlin.math.min

fun selectionSort(unsortedlist:MutableList<Int>):MutableList<Int>{
    val n = unsortedlist.size
    for (i in 0..<n-1){
        var minindex = i
        for (j in i+1..<n){
            if (unsortedlist[j] < unsortedlist[minindex]){
                minindex = j
            }
        }
        val temp = unsortedlist[i]
        unsortedlist[i]= unsortedlist[minindex]
        unsortedlist[minindex] = temp



    }
    return unsortedlist
}

fun main(){
    val unsortedlist = mutableListOf(3,2,9,-4,4,2)
    println(
        selectionSort(unsortedlist)
    )
}

File: python-algo/selection_sort.py
def selection_sort(unsortedlist:list[int])->list[int]:
    n = len(unsortedlist)
    for i in range(0, n-1):
        minindex = i
        for j in range(i+1,n):
            if unsortedlist[j] < unsortedlist[minindex] :
                minindex = j
        temp = unsortedlist[i]
        unsortedlist[i] = unsortedlist[minindex]
        unsortedlist[minindex] = temp
    return unsortedlist

def main():
    unsortedlist = [1, 5, -2, 6, 7, 3, 8, 4]
    print(selection_sort(unsortedlist))

main()

